### ISSUE TITLE
[BUG] Copy to clipboard Sharing does not function

### ISSUE THREAD

**Issue by glottisfaun0000** [FreshRSS/FreshRSS#5595]
**Describe the bug**
The Clipboard feature in Sharing does not function.

**To Reproduce**
1. Go to Configuration > Sharing
2. Add a sharing method > [Clipboard](https://freshrss.github.io/FreshRSS/en/users/08_sharing_services.html)
3. Open an article
4. Click Share > Clipboard
5. Clipboard contents hasn't changed

**Expected behavior**
Clicking Share > Clipboard copies article URL to system clipboard

**Environment information (please complete the following information):**
- Device: Macbook M1
- OS: macOS Ventura 13.4.1
- Browser: Firefox 116.0.2 [64-bit]
- FreshRSS version: 1.21.0 
- Installation type: Docker-compose on macOS Ventura Apple Silicon

Firefox is able to copy from other webpages including self-hoted webapps.

**Comment by math-GH** [comment]
Can you please try to reproduce on demo.freshrss.org?

**Comment by glottisfaun0000** [comment]
> Can you please try to reproduce on demo.freshrss.org?

Working on the demo! Must be something about my setup. Any other information I can provide? Tried with no addons and in Safari and can't copy so I don't think it's the browser.

Docker image: lscr.io/linuxserver/freshrss:latest

**Comment by SCRUMschau** [comment]
Which theme do you use?
Any log entries in the developer tools (mouse menu -> inspect -> console tab)?

**Comment by glottisfaun0000** [comment]
Was on Nord theme, switched theme and issue persists.

As for console:
`Content-Security-Policy: The page’s settings blocked the loading of a resource at inline (“default-src”).`
`The Notification permission may only be requested in a secure context.`

I just do everything local over http with no SSL.

I will note that I have [Copy2Clipboard](https://github.com/cn-tools/cntools_FreshRssExtensions) installed and can successfully copy links from that.

**Comment by math-GH** [comment]
I checked it.

The important line that is doing the clipboard copy is:
https://github.com/FreshRSS/FreshRSS/blob/6749adc05066b42843f803362288f5c1274c1572/p/scripts/main.js#L1091

But it works only in secure context (https or localhost):
`you will need to have a [secure context](https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts) while accessing the Clipboard API. This usually means you can only access this API when you are on localhost or a webpage with a valid SSL/TLS certificate.`
(source: https://phiilu.com/accessing-the-clipboard-in-javascript-using-the-clipboard-api)

I guess that Copy2Clipboard found another way to work with the clipboard

Long story short: in rare cases the clipboard sharing option does not work, but this rare cases are not so unusual for FreshRSS. Let's find a fix.

**Comment by glottisfaun0000** [comment]
Confirmed on my setup that clipboard sharing works via localhost. Thanks for looking into this

**Comment by Frenzie** [comment]
> But it works only in secure context (https or localhost):

Browser makers have purposefully decided to be hostile and they're basically adding everything new as https-only.

In this specific case you could probably still work around it by using the old `document.execCommand('copy')` in an else branch. In short, you need a hidden input element that you select first before you run `document.execCommand('copy')`.

**Comment by math-GH** [comment]
fix is available: #5606 
Feedback and tests are very welcome
